.github/workflows/e2e-browser.yml

```
@@ -0,0 +1,65 @@
+name: End-to-end browser-based tests
+
+on:
+  push:
+
+env:
+  CI: true
+jobs:
+  e2e-browser:
+    # Dependabot cannot access secrets, so it doesn't have a token to authenticate to ESS.
+    # Since all the other jobs of this workflow depend on this one, skipping it should
+    # skip the entire workflow.
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    runs-on: ${{ matrix.os }}
+    environment:
+      name: ${{ matrix.environment-name }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node-version: [16.x]
+        # NSS does not support static client registration, which we rely on for testing.
+        environment-name: ["ESS Dev-Next"]
+    steps:
+      - uses: actions/checkout@v2.4.0
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2.5.1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Cache node modules
+        uses: actions/cache@v2.1.7
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node${{ runner.node-version }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
```

```suggestion
    runs-on: ubuntu-latest
    environment:
      name: ${{ matrix.environment-name }}
    strategy:
      matrix:
        # NSS does not support static client registration, which we rely on for testing.
        environment-name: ["Inrupt Dev-Next", "Inrupt Production"]
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
        with:
          node-version: "16"
          cache: "npm"
```

The setup node action already deals with caching: https://github.com/marketplace/actions/setup-node-js-environment#caching-packages-dependencies.

 Let's not use a matrix when we don't need it (I know it's a mistake we have in other CI, but let's not reproduce it here, one less to fixup ^^,).
 
 Let's maybe use a set of standardised environment names across repos: https://github.com/inrupt/solid-client-js/settings/environments.
 
 I think we should test against Inrupt Production and potentially only make that one gating.
 
 There is literally no interest in making GitHub Actions versions so specific. Workflows will not break because of minor/patch changes and it just creates unnecessary dependabot noise.